src/jit/lsra.cpp

```diff
@@ -5679,11 +5679,12 @@ regNumber LinearScan::tryAllocateFreeReg(Interval* currentInterval, RefPosition*
 
     if (availablePhysRegInterval != nullptr)
     {
-        intervalToUnassign = availablePhysRegInterval->assignedInterval;
         if (isAssigned(availablePhysRegInterval ARM_ARG(currentInterval->registerType)))
         {
             unassignPhysReg(availablePhysRegInterval ARM_ARG(currentInterval->registerType));
-            if (bestScore & VALUE_AVAILABLE)
+            intervalToUnassign = availablePhysRegInterval->assignedInterval;
```

Comment:
You have just unassigned the interval, so this will now be `nullptr`. I think that you need to capture `intervalToUnassign` before calling `unassignPhysReg`